includes/Core/Modules/Modules.php

```diff
@@ -197,6 +199,32 @@ public function __construct(
 		}
 	}
 
+	/**
+	 * Determines if Thank with Google module should be enabled.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool True if the module should be enabled, false otherwise.
+	 */
+	public static function should_enable_twg() {
+		if ( ! Feature_Flags::enabled( 'twgModule' ) ) {
+			return false;
+		}
+
+		if ( Build_Mode::get_mode() === Build_Mode::MODE_DEVELOPMENT ) {
+			return true;
+		}
+
+		if ( ! is_ssl() ) {
+			if ( isset( $_SERVER['HTTP_X_FORWARDED_PROTO'] ) && 'https' === $_SERVER['HTTP_X_FORWARDED_PROTO'] ) {
+				return true;
+			}
+			return false;
+		}
+
+		return true;
```

Comment:
I wonder if returning early here might be a bit more clear, eg.:

```suggestion
		if ( is_ssl() ) {
			return true;
		}

		if ( isset( $_SERVER['HTTP_X_FORWARDED_PROTO'] ) && 'https' === $_SERVER['HTTP_X_FORWARDED_PROTO'] ) {
			return true;
		}

		// Because we aren't in development mode and haven't detected SSL being enabled, TwG should
		// not be enabled.
		return false;
```